### PR TITLE
add manifest submission component 

### PR DESCRIPTION
**Pull request by lakikowolfe**
Use `model/submit` endpoint to upload modified manifest

### CODE REVIEW

**Comment by lakikowolfe** [comment]
Currently blocked: `model/submit` endpoint producing errors.

**Comment by lakikowolfe** [comment]
unblocked and finished.

**Comment by lakikowolfe** [comment]
@milen-sage

**Comment by lakikowolfe** [comment]
related to #5

File: R/app_server.R [hunk 1]
```diff
@@ -29,43 +32,62 @@ app_server <- function( input, output, session ) {
...
+
+    if (!status_col %in% colnames(mani)) {
+      mani$x <- NA
+      names(mani)[names(mani) == "x"] <- status_col
       }
-    
+
+    # add release_status_selection to selected rows
+    mani[file_selection$selected_rows(), status_col] <- release_status_selection$status_selection()
+
```
Comment (andrewelamb) [inline R/app_server.R:52]: I'm having trouble figuring out what this code does.

Comment (andrewelamb) [inline R/app_server.R:52]: I'm having trouble figuring out what this code does.

File: R/app_server.R [hunk 2]
```diff
@@ -29,43 +32,62 @@ app_server <- function( input, output, session ) {
...
-        names(mani)[names(mani) == "x"] <- status_col
-        mani[file_selection$selected_rows(), status_col] <- release_status_selection$status_selection()
-      
+    # if column doesn't exist, add it
+
+    status_col <- "ReleaseStatus"
+
+    if (!status_col %in% colnames(mani)) {
+      mani$x <- NA
+      names(mani)[names(mani) == "x"] <- status_col
```
Comment (andrewelamb) [inline R/app_server.R:47]: Could you comment on what lines 38-47 do?

Comment (andrewelamb) [inline R/app_server.R:47]: Could you comment on what lines 38-47 do?

File: R/app_server.R (hunk 1 above)
Comment (andrewelamb) [inline R/app_server.R:52]: Could you comment on this line does?

Comment (andrewelamb) [inline R/app_server.R:52]: Could you comment on this line does?

File: notes.R [hunk 3]
```diff
@@ -105,34 +105,30 @@ tst_mani <- manifest_download(asset_view = asset_view,
 
 tst_mani_df <- jsonlite::fromJSON(tst_mani)
 
+names(tst_mani_df)[10] <- "HTAN Participant ID"
+tst_mani_df$`Age Is Obfuscated` <- c("False", "False")
+tst_mani_df[1, "Race"] <- "Not Reported"
+
```
Comment (andrewelamb) [inline notes.R:111]: Could you comment on what lines 108 -110 do?

Comment (andrewelamb) [inline notes.R:111]: Could you comment on what lines 108 -110 do?

**Comment by andrewelamb** [comment]
It mostly seems fine to me. I made a few comments where it wasn't clear to me what the code is doing.

**Comment by andrewelamb** [comment]
It mostly seems fine to me. I made a few comments where it wasn't clear to me what the code is doing.

File: R/app_server.R (hunk 1 above)
Comment (lakikowolfe) [reply R/app_server.R:52]:
```
mani[file_selection$selected_rows(), status_col] 
```
Subset manifest by rows selected from `mod_file_selection` and status column

```
release_status_selection$status_selection()
```
Assign to those rows/that column the selected release status from `mod_release_status`

The result is a manifest that has rows in the Release Status column with an updated release status.

File: R/app_server.R (hunk 2 above)
Comment (lakikowolfe) [reply R/app_server.R:47]: Here I am checking the the manifest for if the status_col exists. If it doesn't exist, assign the column names `x` and fill with `NA`. Then assign the `status_col` variable to column `x`.

It's pretty ugly but this was the best way I could figure out to assign a variable to a column name.

File: R/app_server.R (hunk 1 above)
Comment (lakikowolfe) [reply R/app_server.R:52]: repeat question

File: R/app_server.R (hunk 2 above)
Comment (lakikowolfe) [reply R/app_server.R:47]: repeat question

File: notes.R (hunk 3 above)
Comment (lakikowolfe) [reply notes.R:111]: repeat question

File: R/app_server.R (hunk 2 above)
Comment (andrewelamb) [reply R/app_server.R:47]: If you are using dplyr you can use this one liner: mani <- dplyr::mutate(mani, !!status_col := NA)

Comment (lakikowolfe) [reply R/app_server.R:47]: Oh I like that!

**Pull request merged**